client/app/services/Download.js

```diff
@@ -0,0 +1,17 @@
+(function (angular) {
+	"use strict";
+
+	var module = angular.module('myCloudDriveApp');
+
+	function Download($stateParams) {
+
+        return {
+            download: function(fileName) {
+                var path = '/files/download' + $stateParams.path + '/' + fileName;
+                window.open(path, '_self');
```

Comment:
It is a bad practice to use a global window object. Angular's $window service refers to it.